torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1905,14 +2190,21 @@ def hint():
                 # won't matter because sizes in practice will be no where near
                 # the 64-bit limit.
                 if r.upper != sympy.oo and r.upper < sys.maxsize - 1:
+                    if any(is_dim(source) for source in sources):
+                        dim_constraints.add(sympy.Le(symbol, r.upper))
                     bounds.append(str(r.upper))
                 if len(bounds) > 1:
                     exprs.append(" <= ".join(bounds))
 
+        if torch._dynamo.config.dynamic_shapes and torch._dynamo.config.summarize_dim_constraints:
+            dim_constraints.solve()
+            log.warning(f"Summary of dimension constraints:{dim_constraints.prettify_results()}")
```

Comment:
No dynamo config notions in this file please. We already have dynamo source notions here in some places and it kills me.

Reply:
How would you suggest I do this, then, assuming I do want to gate it with a flag?

Reply:
Well, the other way we've been doing it is passing these in inside `__init__`. BTW there's no point in testing `dynamic_shapes`, you'll never hit ShapeEnv without it True.